src/main/java/com/allmycoins/balance/osmosis/OsmosisProvider.java

```diff
@@ -0,0 +1,142 @@
+package com.allmycoins.balance.osmosis;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import com.allmycoins.balance.PublicAddressBalanceProvider;
+import com.allmycoins.balance.cosmosjs.CosmosJsBalanceJson;
+import com.allmycoins.balance.cosmosjs.CosmosJsProvider;
+import com.allmycoins.json.BalanceJson;
+import com.allmycoins.utils.RequestUtils;
+
+public final class OsmosisProvider implements PublicAddressBalanceProvider {
+
+	private static final String TOKEN = "OSMO";
+
+	/* From https://docs.osmosis.zone/osmosis-core/asset-info/. */
+	private static final Map<String, String> IBC_DENOM_TO_SYMBOL_MAP = new HashMap<>();
+
+	static {
+		IBC_DENOM_TO_SYMBOL_MAP.put("uosmo", "OSMO");
+		IBC_DENOM_TO_SYMBOL_MAP.put("ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2", "ATOM");
```

Comment:
We can also get the list of assets from
https://raw.githubusercontent.com/osmosis-labs/assetlists/main/osmosis-1/osmosis-frontier.assetlist.json